packages/flutter/lib/src/services/text_input.dart

```diff
@@ -814,6 +826,9 @@ class _TextInputClientHandler {
       case 'TextInputClient.updateFloatingCursor':
         _currentConnection._client.updateFloatingCursor(_toTextPoint(_toTextCursorAction(args[1]), args[2]));
         break;
+      case 'TextInputClient.onConnectionClosed':
+        _currentConnection._client.connectionClosed();
```

Comment:
How does this not suffer from the same bugs that led to reverting https://github.com/flutter/flutter/pull/39523/files

Reply:
We don't think this will break android since:

(1) This is an optional channel, we will use it on web, android won't get effected as long as it is not implemented. 
(2) The implementations are not the same. In the previous implementation the connection was getting closed but the editabletext widget's state was not changed. In this implementation as well as closing the connection, these two lines are added in editabletext:
_lastKnownRemoteTextEditingValue = null;
 _finalizeEditing(true);